$NetBSD$

* getaddrinfo(3) is in XNS5.2, require _XOPEN_SOURCE>=520
* _NETBSD_SORCE is required for NI_MAXHOST/NI_MAXSERVE on NetBSD.
* Look for a more useful SunOS macro

--- src/dbmail.h.in.orig	2012-03-11 08:02:30.000000000 +0000
+++ src/dbmail.h.in
@@ -32,13 +32,16 @@
 
 #define __EXTENSIONS__ /* solaris */
 
-#if defined(__FreeBSD__) || defined(__APPLE__) || defined(__SUNPRO_C)
+#if defined(__FreeBSD__) || defined(__APPLE__) || defined(__sun) || defined(__NetBSD__)
 
-#define _XOPEN_SOURCE
+#define _XOPEN_SOURCE	520
+#ifdef __NetBSD__
+#define _NETBSD_SOURCE
+#endif
 
 #else
 
-#define _XOPEN_SOURCE	500
+#define _XOPEN_SOURCE	520
 #include <features.h>
 
 #endif
